pkg/agent/datamodel/types.go

```diff
@@ -1186,6 +1190,13 @@ func (a *AgentPoolProfile) IsAvailabilitySets() bool {
 	return strings.EqualFold(a.AvailabilityProfile, AvailabilitySet)
 }
 
+func truncateString(str string, length int) string {
+	if len(str) > length {
+		return str[:length]
+	}
+	return str
+}
+
 // GetKubernetesLabels returns a k8s API-compliant labels string for nodes in this profile.
 func (a *AgentPoolProfile) GetKubernetesLabels() string {
```

Comment:
this is too risky I find here, we should instead truncate when we create the label since this Label comes from US, it's tied to the customHeader.  Client side should block it. or since we are dynamically creaeting it, I feel we should split the long label into multiple labels.

Reply:
Have changed the PR to reject long labels rather than truncate them.

Reply:
Changed to reject requests.